src/Core/Binary.idr

```diff
@@ -160,8 +160,11 @@ HasNames e => HasNames (TTCFile e) where
           = pure $ Just $ MkRewriteNs !(resolved gam e) !(resolved gam r)
 
       resolvedPrim : Context -> PrimNames -> Core PrimNames
-      resolvedPrim gam (MkPrimNs mi ms mc)
-          = pure $ MkPrimNs !(resolved gam mi) !(resolved gam ms) !(resolved gam mc)
+      resolvedPrim gam (MkPrimNs mi ms mc md)
+          = pure $ MkPrimNs !(resolved gam mi)
+                            !(resolved gam ms)
+                            !(resolved gam mc)
+                            !(resolved gam md)
```

Comment:
The the same note as for `fullPrim` applies here.

Reply:
Well, personally, I find this version more appealing in this case, since the idiom brackets will not fit on a single line, so we do - in my opinion - not really reduce the visual noise here. But if there are strong reasons to change this, I'll be happy to include them.

Reply:
Well, these two forms definitely compile to different code (i.e. `>>=` is not used when idiom brackets are used). But since it was previously defined with !-syntax, there are no strong reasons to change it.

Reply:
It'd be a better refactoring to define `record PrimNames' (a : Type)` and
then have `PrimNames = PrimNames Name` so that you may define a `Traversable`
instance for `PrimNames'` and replace all of these `MkPrimNs !(f ...) !(f ...)`
with simply `traverse f primNs`.

Reply:
gallais, that doesn't work here since `Core` seems not to have an `Applicative` instance. Comments in `Core.idr` suggest that this is due to performance reasons. Therefore, I'm not sure whether this is worth the hassle as implementating `Traversable` for `PrimNames'` is quite verbose itself, since the most straightforward implementation using idiom brackets results in ambiguity errors. Still, if you think this will pay out in the long run, I'll change `PrimNames` as suggested.

Reply:
I don't think it's necessary. I was just saying I'd rather see that
than a change from `MkP !(...) !(...)` to `[| MkP (...) (...) |]`.